### ISSUE TITLE
Fix attention meta/fake registrations & tests for seed device

### ISSUE THREAD

**Issue by davidberard98** [pytorch/pytorch#121528]
### 🐛 Describe the bug

https://github.com/pytorch/pytorch/pull/120839 changed the fake_impls to put seed/offset on CPU instead of CUDA. But this is only true when we're not using CUDA graphs

https://github.com/pytorch/pytorch/blob/df06b94778cb65c29b1b606d556ef0c8352f9e59/aten/src/ATen/native/transformers/cuda/attention.cu#L1018-L1024

That PR broke cudagraphs + SDPA (tests don't exist yet, but @Chillee provided this repro):
```python
import torch
from torch import nn
import torch.nn.functional as F
import torch.autograd as autograd
torch.set_default_device('cuda')

q = torch.randn(8, 128, 1024, 128)
k = torch.randn(8, 128, 1024, 128)
v = torch.randn(8, 128, 1024, 128)

@torch.compile(mode="reduce-overhead")
def f(q, k, v):
    return F.scaled_dot_product_attention(q, k, v)

for _ in range(2):
    f(q, k, v)
with torch.profiler.profile() as prof:
    for _ in range(5):
        f(q, k, v)
prof.export_chrome_trace("/home/dberard/sdpa.json")
```

The PR was reverted, so we should figure out what to do with the meta registration now

TODO:
- [ ] Maybe move the fake impl back to meta_registrations, if we decide to just leave the seed device as `cuda`. (There was a PR to move the attn kernels from meta_registrations to fake_impls (https://github.com/pytorch/pytorch/pull/120682) but there was also a PR on top of that to fix max_seqlen_q for nested tensors (https://github.com/pytorch/pytorch/pull/120842))
  * [ ] If so, let's also maybe add a special case in test_fake so that it passes for sdpa / efficient_attention / flash_attention
  * [ ] Unskip the test hopefully
  * [ ] Add a bit more comments
- [ ] Alternatively, see if we can make the fake impl return "cuda" for cudagraphs and "cpu" otherwise?
  * This might not actually be possible, I'm guessing fake impl is collected before cudagraph tracing, so we won't know at that point whether the goal is to use cudagraphs or not.
- [ ] Add the repro above as a test somewhere 

### Versions

main branch, march 6

cc @ezyang @msaroufim @bdhirsh @anijain2305 @zou3519 @chauhang

**Comment by davidberard98** [comment]
sort of fixed in #134288 - moved kernels back to meta_registrations and re-enabled the opinfos that we were targeting when we moved the tests to fake_impls, by skipping the specific part of the opinfo test that was failing.
